Programs/brlapi_client.c

```diff
@@ -1659,12 +1660,33 @@ static int getControllingTty(void)
 
 #ifdef linux
   {
-    FILE *stream;
+    int fd;
+    if ((fd = open("/proc/self/stat", O_RDONLY, 0)) >= 0) {
+      char buf[256];
+      size_t read_bytes = read(fd, buf, sizeof(buf) - 1);
```

Comment:
`read` returns an `ssize_t`, better use that, and check that it suceeded.

To simplify the whole thing, maybe use a `do { } while(0)` block around the whole thing, so you can just `break` in error cases

Reply:
Done